core/rfb.js

```diff
@@ -1325,18 +1364,144 @@ export default class RFB extends EventTargetMixin {
+                if (this._clipboardText != null &&
+                    this._clipboardServerCapabilitiesActions[extendedClipboardActionProvide]) {
+
+                    if (formats & extendedClipboardFormatText) {
+                        RFB.messages.extendedClipboardProvide(this._sock, [extendedClipboardFormatText], [this._clipboardText]);
+                    }
+                }
+
+            } else if (actions == extendedClipboardActionPeek) {
+                if (this._clipboardServerCapabilitiesActions[extendedClipboardActionNotify]) {
+
+                    if (this._clipboardText != null) {
+                        RFB.messages.extendedClipboardNotify(this._sock, [extendedClipboardFormatText]);
+                    } else {
+                        RFB.messages.extendedClipboardNotify(this._sock, []);
+                    }
+                }
+
+            } else if (actions == extendedClipboardActionNotify) {
+                if (this._clipboardServerCapabilitiesActions[extendedClipboardActionRequest]) {
+
+                    if ((formats & extendedClipboardFormatText) == extendedClipboardFormatText) {
+                        RFB.messages.extendedClipboardRequest(this._sock, [extendedClipboardFormatText]);
+                    }
+                }
+
+            } else if (actions == extendedClipboardActionProvide) {
+                if ((formats & extendedClipboardFormatText) != extendedClipboardFormatText) {
+                    return true;
+                }
+                // Ignore what we had in our clipboard client side.
+                this._clipboardText = null;
+
+                // FIXME: Should probably verify that this data was actually requested
+                let zlibStream = this._sock.rQshiftBytes(length - 4);
+                let streamInflator = new Inflator();
+                let textData = null;
+
+                streamInflator.setInput(zlibStream);
+                for (let i = 0; i <= 15; i++) {
+                    let chunk;
+
+                    if ((formats >> i) & 1) {
+
+                        let size = 0x00;
+                        let sizeArray = streamInflator.inflate(4);
+
+                        size |= (sizeArray[0] << 24);
+                        size |= (sizeArray[1] << 16);
+                        size |= (sizeArray[2] << 8);
+                        size |= (sizeArray[3]);
+                        chunk = streamInflator.inflate(size);
+                    }
 
+                    if ((((1 << i) & formats) == extendedClipboardFormatText) && (chunk != null)) {
+                        textData = String.fromCharCode.apply(null, chunk);
+
+                        textData = decodeUTF8(textData);
+                        if ((textData.length > 0) && "\0" == textData.charAt(textData.length - 1)) {
+                            textData = textData.slice(0, -1);
+                        }
+
+                        textData =textData.replace("\r\n", "\n");
```

Comment:
There's a missing space here.

Reply:
this seems like something eslint should have cought